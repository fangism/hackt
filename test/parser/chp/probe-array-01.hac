// "probe-array-01.hac"
// expect fail: probe statements should expect scalar channels
// reported by Carlos Tadeo Ortega Otero

defproc a (chan?(int) INPUT[2]) {
	int x;
	chp {
	*[
		[#INPUT -> INPUT[0]?(x)	// bad probe!
		[]else -> skip
		]
	]
	}
}

a b;

