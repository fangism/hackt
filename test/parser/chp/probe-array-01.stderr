ERROR expecting a scalar channel reference, but got a 1-dim reference.  [9:5..9]
ERROR in guard expression at [9:4..9]
At least one error in guarded statement list in [9:3--11:3]
ERROR in one of the actions in [8:2--12:2]
ERROR: at least one error in CHP body.
